subsys/pm/device_runtime.c

```diff
@@ -576,3 +576,33 @@ int pm_device_runtime_usage(const struct device *dev)
 
 	return usage;
 }
+
+int pm_device_runtime_flush(const struct device *dev)
+{
+	struct pm_device *pm = dev->pm;
+	int ret = 0;
+
+	if (!pm_device_runtime_is_enabled(dev)) {
+		return -ENOTSUP;
+	}
+
+	(void)k_sem_take(&pm->lock, K_FOREVER);
+
+	if (pm->base.state == PM_DEVICE_STATE_SUSPENDED) {
+		ret = -EALREADY;
+		goto unlock;
+	}
+
+	if ((pm->base.state == PM_DEVICE_STATE_SUSPENDING) &&
+	    ((k_work_cancel_delayable(&pm->work) & K_WORK_RUNNING) == 0)) {
```

Comment:
Any conditional logic around `k_work_*` always makes me nervous due to the wall of text about it in the docs: https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html#avoid-race-conditions

Could this be implemented by calling `k_work_cancel_delayable_sync` and then performing any checks/actions?